operators/src/pro/ml/xgboost.rs

```diff
@@ -0,0 +1,628 @@
+            data_type: RasterOut,
+            time: in_desc.time,
+            bbox: in_desc.bbox,
+            resolution: Some(fin_res), // QUESTION: is this allowed to be none, or is a value required?
+            spatial_reference: in_desc.spatial_reference,
+
+            // where do we get this information from?
+            // test_data/dataset_defs? If so, how to load it?
+            measurement: Measurement::Classification(ClassificationMeasurement {
+                measurement: "raw".into(),
+                classes: hashmap!(0 => "Water Bodies".to_string()),
+            }),
+        };
+        let initialized_operator = InitializedXgboostOperator {
+            result_descriptor: out_desc,
+            sources: init_rasters,
+            model_file_path: self.params.model_file_path,
+            no_data_value: self.params.no_data_value,
+        };
+
+        Ok(initialized_operator.boxed())
+    }
+
+    span_fn!(XgboostOperator);
+}
+
+impl InitializedRasterOperator for InitializedXgboostOperator {
+    fn result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> Result<TypedRasterQueryProcessor> {
+        let vec_of_rqps: Vec<Box<dyn RasterQueryProcessor<RasterType = f32>>> = self
+            .sources
+            .iter()
+            .map(|init_raster| init_raster.query_processor().unwrap().into_f32())
+            .collect();
+
+        Ok(QueryProcessorOut(Box::new(XgboostProcessor::new(
+            vec_of_rqps,
+            self.model_file_path.clone(),
+            self.no_data_value,
+        ))))
+    }
+}
+
+struct XgboostProcessor<Q, P>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+{
+    sources: Vec<Q>,
+    model_file: PathBuf,
+    no_data_value: Option<f32>,
+}
+
+impl<Q, P> XgboostProcessor<Q, P>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+    P: Pixel,
+{
+    pub fn new(sources: Vec<Q>, model_file_path: PathBuf, no_data_value: Option<f32>) -> Self {
+        Self {
+            sources,
+            model_file: model_file_path,
+            no_data_value,
+        }
+    }
+
+    async fn process_tile_async(
+        &self,
+        mut bands_of_tile: Vec<Result<RasterTile2D<P>, crate::error::Error>>,
+        pool: Arc<ThreadPool>,
+    ) -> Result<RasterTile2D<PixelOut>> {
+        let tile = bands_of_tile.pop().ok_or(error::Error::EmptyInput)??;
+
+        // gather the data
+        let grid_shape = tile.grid_shape();
+        let n_bands = bands_of_tile.len() as i32 + 1; // +1 because of .pop()
+        let props = tile.properties.clone(); // = &tile.properties;
+        let time = tile.time;
+        let tile_position = tile.tile_position;
+        let global_geo_transform = tile.global_geo_transform;
+
+        let model = self.model_file.clone();
+        let ndv = self.no_data_value.or(Some(f32::NAN));
+
+        let predicted_grid: geoengine_datatypes::raster::Grid<GridShape<[usize; 2]>, f32> =
```

Comment:
you can also use the `Grid2d` type here

Reply:
https://github.com/geo-engine/geoengine/pull/639/commits/5815f849de667bc249ec87211fc8ecf941a99970